src/image.rs

```diff
@@ -463,6 +463,11 @@ where
     let mut tmp_scanline = None;
 
     {
+        progress_callback(Progress {
+            current: 0,
+            total: total_bytes,
+        });
+
```

Comment:
I'm not sure about the recursion you mentioned. But, as the non-mutable callback could be referenced multiple times, but the mutable callback can only be referenced once, it can be used in less situations now. That's why I had to change the order of this call, as it was captured later on, which would attempt to reference the mutable callback twice.